### ISSUE TITLE
no-unused-variable: variable used in decorator function throws error

### ISSUE THREAD

**Issue by dms1lva** [palantir/tslint#1421]
### Bug Report
- **TSLint version**: `3.13.0`
- **TypeScript version**: `1.8.10`
- **Running TSLint via**: CLI
#### TypeScript code being linted

``` ts
class NiceClass {
  @myNiceDecorator()
  public methodWithDecorator() {

  }

  private methodUsedInDecorator() {

  }
}

function myNiceDecorator() {
  this.methodUsedInDecorator();
}
```

with `tslint.json` configuration:

``` json
{
     "rules" :{
         "no-unused-variable": true
     }
}
```
#### Actual behavior

This error is thrown:
random.ts[7, 11]: Unused method: 'methodUsedInDecorator'
#### Expected behavior

No error should be thrown

**Comment by andy-hanson** [comment]
Should be fixed by #2235.
